packages/editor/src/components/block-settings-menu/content-only-settings-menu.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	BlockSettingsMenuControls,
+	__unstableBlockSettingsMenuFirstItem as BlockSettingsMenuFirstItem,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { store as coreStore } from '@wordpress/core-data';
+import { __experimentalText as Text, MenuItem } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+function ContentOnlySettingsMenuItems( { clientId } ) {
+	const { entity, onNavigateToEntityRecord } = useSelect(
+		( select ) => {
+			const {
+				getBlockEditingMode,
+				getBlockParentsByBlockName,
+				getSettings,
+				getBlockAttributes,
+			} = select( blockEditorStore );
+			const contentOnly =
+				getBlockEditingMode( clientId ) === 'contentOnly';
+			if ( ! contentOnly ) return {};
+			const patternParent = getBlockParentsByBlockName(
+				clientId,
+				'core/block',
+				true
+			)[ 0 ];
+
+			let record;
+			if ( patternParent ) {
+				record = select( coreStore ).getEntityRecord(
+					'postType',
+					'wp_block',
+					getBlockAttributes( patternParent ).ref
+				);
+			} else {
+				const templateId = select( editorStore ).getCurrentTemplateId();
+				if ( templateId ) {
+					record = select( coreStore ).getEntityRecord(
+						'postType',
+						'wp_template',
+						templateId
+					);
+				}
+			}
+			return {
+				entity: record,
+				onNavigateToEntityRecord:
+					getSettings().onNavigateToEntityRecord,
+			};
+		},
+		[ clientId ]
+	);
+
+	if ( ! entity ) return null;
+
+	const isPattern = entity.type === 'wp_block';
+
+	return (
+		<>
+			<BlockSettingsMenuFirstItem>
+				<Text
+					variant="muted"
+					as="p"
+					className="editor-content-only-settings-menu__description"
+				>
+					{ isPattern
+						? sprintf(
+								// translators: %s: pattern's title.
+								__(
+									'This block belongs to "%s". Edit the pattern to move or delete it.'
+								),
+								entity.title.raw
+						  )
+						: sprintf(
+								// translators: %s: template's title.
+								__(
+									'This block belongs to "%s". Edit the template to move or delete it.'
+								),
+								entity.title.rendered
+						  ) }
```

Comment:
It is possible to have a `contentOnly` locked group block (or just about any other block) that's not a pattern or a template, and this seems like an unhandled case.

To repro:
1. Add a group block with some inner blocks
2. Go to Code Editor mode and add `"templateLock": "contentOnly"` to the group block's attributes.
3. Return to Visual Editor mode and open the block options menu in List View, observe it mentions being part of a template, even thought that's not the case.

Reply:
Is it a case only when the users manually edit the markup in the code editor? I wonder if we should just hide the menu button until we find something to show there? c.c. @jasmussen for any design feedback.

Reply:
Thanks for the context, and ping. I might need even more context, but let me try and restate my understanding. First, here's a pattern I inserted manually. But I might as well have just typed this out, grouped it, built the pattern myself:
![plain group with stuff](https://github.com/WordPress/gutenberg/assets/1204802/04ce55ee-7158-4ae9-b6bf-81db60dd9897)

If I go into the editor, and add `"templateLock": "contentOnly"` to the group block, it becomes, well, contentOnly locked:

![menu](https://github.com/WordPress/gutenberg/assets/1204802/b765b6f2-8862-4b34-a21d-e9e2f6b29070)

To an extent, this is indeed an unhandled case, because right now there's no way to apply that property to a group without manually editing. IMO that _should_ be possible, because contentOnly locking is a valuable aspect of the editor, something that benefits not just patterns or "sections". As you can see, I'm not seeing that ellipsis menu, though. Did you already hide it, as you suggested, or am I missing a beat?

Hiding it seems fine, at least for now. There's still the larger question: how do I enable contentOnly locking, and how do I turn it off? IMO it could be part of the "Lock" modal, and can be handled separately. CC: @jameskoster

Reply:
> As you can see, I'm not seeing that ellipsis menu, though. Did you already hide it, as you suggested, or am I missing a beat?

Yep, I just pushed [8b4ae5e](https://github.com/WordPress/gutenberg/pull/61127/commits/8b4ae5eda2d3c3d2faca8cb276eebb7d227e0856) to hide the menu in this case 😅.

Reply:
Hmm good discussion.

Hiding may be okay in the short term, but on balance I think I'd lean towards not hiding for a couple of reasons: it gives no explanation about _why_ there's no ellipsis button when you might expect to find one; it's also inconsistent with the other cases.

Couldn't we employ the same pattern we're using in pages / synced patterns and show a contextual message in the menu on click. Something like this, but worded better:

> Only the content of blocks inside "Group" can be edited

There might be a menu item to remove content lock?

Reply:
I pushed [4d37b10](https://github.com/WordPress/gutenberg/pull/61127/commits/4d37b1029d2f216980a5d279a905f75fccff209f) to show the suggested description and menu item.

![image](https://github.com/WordPress/gutenberg/assets/7753001/58261ac2-513a-4a01-9265-b6106296ea06)

Upon clicking the "Remove template lock" menu item, the parent (Group in this case) block will remove the "templateLock" attribute. Not sure if it's the desired action.

Not too happy about the wording either 😅. I could use some suggestions. Please feel free to push commits directly too!

Reply:
Yeah the wording is tricky... I wonder if it would work better the other way around, IE 

> The parent 'Group' block is template-locked, preventing the movement or deletion of child blocks, as well as the addition of any inner blocks.

It's a bit of a mouthful, but maybe that's okay given it's a bit of an edge case at the moment? We probably need a better phrase than "template-locked"... it's not intuitive at all. Maybe...

> The parent 'Group' block is partially locked, preventing the movement or deletion of child blocks, as well as the addition of any inner blocks.
> Unlock 'Group'

Reply:
Sure! I pushed [f04744b](https://github.com/WordPress/gutenberg/pull/61127/commits/f04744bebd29baeb1aec5b86f70131b13c3aa18c) for this rewording.

![image](https://github.com/WordPress/gutenberg/assets/7753001/d6719d24-8f81-4095-9609-5ce626f20b16)

Reply:
I reckon this covers the different cases reasonably well, and certainly improves on the lock icon.

I could go either way on the "Unlock..." menu item. It's a little odd to have a UI to disable a feature, but not one to re-enable it. That could potentially be explored separately.

Reply:
> It's a little odd to have a UI to disable a feature, but not one to re-enable it.

Yep, I agree. Though it's also the case for some common actions like "Group" etc. One could always use the Undo button if they really want to.

Reply:
We might be able to reuse the existing 'Modify' feature for contentOnly parent blocks.

Reply:
In [0147d37](https://github.com/WordPress/gutenberg/pull/61127/commits/0147d37fb8dd8cb55167a33b7f88da8edcdfe978), I updated it to use the "Modify" button as shown in the recordings:


https://github.com/WordPress/gutenberg/assets/7753001/f7379c6a-36e1-425b-b0b2-68748e87648c